config/software/berkshelf.rb

```diff
@@ -21,14 +21,10 @@
 
 relative_path "berkshelf"
 
-if windows?
-  dependency "ruby-windows"
-  dependency "ruby-windows-devkit"
-  dependency "rubygems"
-else
-  dependency "libffi"
-  dependency "ruby"
-  dependency "rubygems"
+dependency "ruby"
+dependency "rubygems"
+
+unless windows? && (project.overrides[:ruby].nil? || project.overrides[:ruby][:version] == "ruby-windows")
```

Comment:
This is a bit convoluted due to the "unless" but it's a "please build libarchive if we're on non-windows or we're on windows and ruby's version has been overridden from 'ruby-windows' which is the special flag to signify rubyinstaller devkit builds"